src/main/java/cn/nukkit/Server.java

```diff
@@ -2013,6 +1939,58 @@ private void registerBlockEntities() {
         BlockEntity.registerBlockEntity(BlockEntity.MUSIC, BlockEntityMusic.class);
     }
 
+    private void loadLevels()   {
+        Map<String, Object> worldNames = this.getConfig("worlds", Collections.emptyMap());
+        if (worldNames.isEmpty())   {
+            throw new IllegalStateException("No worlds configured! Add a world to nukkit.yml and try again!");
+        }
+        List<CompletableFuture<Level>> levelFutures = new ArrayList<>(worldNames.size());
+
+        for (String name : worldNames.keySet()) {
+            //fallback to level name if no seed is set
+            Object seedObj = this.getConfig("worlds." + name + ".seed", name);
+            long seed;
+            if (seedObj instanceof Number)   {
+                seed = ((Number) seedObj).longValue();
+            } else if (seedObj instanceof String)   {
+                if (seedObj == name)    {
+                    log.warn("World \"{}\" does not have a seed! Using a the name as the seed", name);
+                }
+
+                //this internally generates an MD5 hash of the seed string
+                UUID uuid = UUID.nameUUIDFromBytes(((String) seedObj).getBytes(StandardCharsets.UTF_8));
+                seed = uuid.getMostSignificantBits() ^ uuid.getLeastSignificantBits();
+            } else {
+                throw new IllegalStateException("Seed for world \"" + name + "\" is invalid: " + (seedObj == null ? "null" : seedObj.getClass().getCanonicalName()));
+            }
+
+            levelFutures.add(this.loadLevel().id(name).seed(seed)
+                    .generator(Identifier.fromString(this.getConfig(
```

Comment:
This looks a bit messy. Maybe the identifiers could be created outside of the builder?